ports/analog/supervisor/port.c

```diff
@@ -0,0 +1,319 @@
+ *
+ * Copyright (C) 2022-2023 Maxim Integrated Products, Inc. All Rights Reserved.
+ * (now owned by Analog Devices, Inc.),
+ * Copyright (C) 2023 Analog Devices, Inc. All Rights Reserved. This software
+ * is proprietary to Analog Devices, Inc. and its licensors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+/**
+ * @file port.c
+ * @author Brandon Hurst @ Analog Devices, Inc.
+ * @brief Functions required for a basic CircuitPython port
+ * @date 2024-07-30
+ *
+ * @copyright Copyright (c) 2024
+ */
+
+#include <stdint.h>
+#include <stdio.h>
+#include "supervisor/background_callback.h"
+#include "supervisor/board.h"
+#include "supervisor/port.h"
+
+#include "common-hal/microcontroller/Pin.h"
+#include "shared-bindings/microcontroller/__init__.h"
+
+// Sys includes
+#include "max32_port.h"
+
+// Timers
+#include "mxc_delay.h"
+#include "rtc.h"
+
+// msec to RTC subsec ticks (4 kHz)
+/* Converts a time in milleseconds to equivalent RSSA register value */
+#define MSEC_TO_SS_ALARM(x) (0 - ((x * 4096) / 1000))
+
+// Externs defined by linker .ld file
+extern uint32_t _stack, _heap, _estack, _eheap;
+extern uint32_t _ebss;
+
+// From boards/$(BOARD)/board.c
+extern const mxc_gpio_cfg_t pb_pin[];
+extern const int num_pbs;
+extern const mxc_gpio_cfg_t led_pin[];
+extern const int num_leds;
+
+// For saving rtc data for ticks
+static uint32_t subsec, sec = 0;
+static uint32_t tick_flag = 0;
+
+// defined by cmsis core files
+extern void NVIC_SystemReset(void) NORETURN;
+
+safe_mode_t port_init(void) {
+    int err = E_NO_ERROR;
+
+    // 1ms tick timer
+    SysTick_Config(SystemCoreClock / 1000);
+    NVIC_EnableIRQ(SysTick_IRQn);
+
+    // Enable GPIO (enables clocks + common init for ports)
+    for (int i = 0; i < MXC_CFG_GPIO_INSTANCES; i++) {
+        err = MXC_GPIO_Init(0x1 << i);
+        if (err) {
+            return SAFE_MODE_PROGRAMMATIC;
+        }
+    }
+
+    // Init Board LEDs
+    /* setup GPIO for the LED */
+    for (int i = 0; i < num_leds; i++) {
+        // Set the output value
+        MXC_GPIO_OutClr(led_pin[i].port, led_pin[i].mask);
+
+        if (MXC_GPIO_Config(&led_pin[i]) != E_NO_ERROR) {
+            return SAFE_MODE_PROGRAMMATIC;
+        }
+    }
+
+    // Turn on one LED to indicate Sign of Life
+    MXC_GPIO_OutSet(led_pin[2].port, led_pin[2].mask);
```

Comment:
Did you get the status LED working? It should replace this code.

Reply:
Yep! Removed this code in the latest commit.